File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/tool.py
from __future__ import annotations
from typing import Literal
from pydantic import BaseModel


class ToolMeta(BaseModel):
    name: str
    display_name: str
    category: str
    domain: str
    provider: str
    action_type: Literal["read", "write", "dangerous"]
    risk_level: Literal["low", "medium", "high", "critical"]
    approval_required: bool
    timeout_seconds: int
    idempotent: bool
    version: str
    tags: list[str] = []


class ToolHealthStatus(BaseModel):
    name: str
    provider: str
    healthy: bool
    last_check: str
    latency_ms: int

File: opspilot-enterprise/packages/shared-types/src/tool.ts
export type ActionType = "read" | "write" | "dangerous";

export interface ToolMeta {
  name: string;
  display_name: string;
  category: string;
  domain: string;
  provider: string;
  action_type: ActionType;
  risk_level: "low" | "medium" | "high" | "critical";
  approval_required: boolean;
  timeout_seconds: number;
  idempotent: boolean;
  version: string;
  tags: string[];
}

export interface ToolHealthStatus {
  name: string;
  provider: string;
  healthy: boolean;
  last_check: string;
  latency_ms: number;
}
